data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/grouping/S3Group.java

```diff
@@ -44,7 +44,8 @@ public void releaseEventHandles(final boolean result) {
         groupEventHandles.clear();
     }
 
-    Collection<EventHandle> getGroupEventHandles() {
-        return groupEventHandles;
+    @Override
+    public int compareTo(final S3Group o) {
+        return Long.compare(o.getBuffer().getSize(), buffer.getSize());
```

Comment:
I feel this is a little counter-intuitive that largerGroup.compareTo(smallerGroup) results in a negative value. But it's a minor thing.

Reply:
Yeah it is a little odd for sure. But it means that sort() will give the reverse order, which is what we want.